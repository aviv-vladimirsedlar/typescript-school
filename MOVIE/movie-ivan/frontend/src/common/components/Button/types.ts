export interface ButtonProps {
  className?: string;
  label: string;
  onClick: () => void;
  type?: 'button' | 'submit' | 'reset';
}
