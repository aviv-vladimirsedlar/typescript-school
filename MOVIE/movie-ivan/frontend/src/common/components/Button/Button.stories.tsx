import type { Meta, StoryObj } from '@storybook/react';
import { fn } from '@storybook/test';

// import { ButtonProps } from './types';

import Button from '.';

const meta = {
  title: 'Common/Components/Button',
  component: Button,
  parameters: {},

  tags: ['autodocs'],
  argTypes: {},
  args: { onClick: fn() },
} satisfies Meta<typeof Button>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: { children: 'Default' },
};

export const Primary: Story = {
  args: { children: 'Primary', className: 'bg-blue-500 text-white' },
};

export const Secondary: Story = {
  args: { children: 'Secondary', className: 'bg-gray-500 text-white' },
};

export const Outline: Story = {
  args: { children: 'Outline', className: 'bg-transparent border border-gray-600 text-gray-600' },
};
