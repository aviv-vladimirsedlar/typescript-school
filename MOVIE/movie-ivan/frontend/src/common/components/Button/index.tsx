import clsx from 'clsx';
import React from 'react';

import { ButtonProps } from './types';

const Button: React.FC<ButtonProps> = ({ id, className, children, onClick, type = 'button' }) => (
  <button
    id={id}
    className={clsx('rounded-lg bg-blue-500 p-4 font-bold text-white', className)}
    onClick={onClick}
    type={type}
  >
    {children}
  </button>
);

export default Button;
