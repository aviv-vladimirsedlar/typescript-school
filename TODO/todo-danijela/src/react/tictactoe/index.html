<!DOCTYPE html>
<html>
<body>
  <div id="root"></div>
</body>
<script src="https://unpkg.com/@babel/standalone/babel.min.js"></script>
<script async src="https://ga.jspm.io/npm:es-module-shims@1.7.0/dist/es-module-shims.js"></script>
<script type="text/javascript" src="index.js"/>
</html>
