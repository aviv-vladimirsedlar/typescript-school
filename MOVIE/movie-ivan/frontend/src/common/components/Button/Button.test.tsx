import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';

import '@testing-library/jest-dom';
import Button from '.';

describe('Button component', () => {
  it('should match snapshot when initially rendered', () => {
    const { asFragment } = render(
      <Button data-testid="button" className="p-2" onClick={() => {}}>
        Click Me
      </Button>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the button with correct text', () => {
    render(
      <Button data-testid="button" onClick={() => {}}>
        Click Me
      </Button>,
    );
    expect(screen.getByTestId('button')).toHaveTextContent('Click Me');
  });

  it('applies the correct default classes', () => {
    render(<Button data-testid="button">Styled Button</Button>);
    const button = screen.getByTestId('button');
    expect(button).toHaveClass('rounded-lg bg-blue-500 p-2 px-6 font-semibold text-white');
  });

  it('calls onClick when button is clicked', () => {
    const handleClick = jest.fn();
    render(
      <Button data-testid="button" onClick={handleClick}>
        Click Me
      </Button>,
    );
    fireEvent.click(screen.getByTestId('button'));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
