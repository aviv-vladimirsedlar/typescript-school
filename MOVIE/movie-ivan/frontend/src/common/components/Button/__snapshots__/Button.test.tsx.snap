// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match snapshot when initially rendered 1`] = `
<DocumentFragment>
  <button
    class="rounded-lg bg-blue-500 p-2 px-6 font-semibold p-2 text-white"
    data-testid="button"
    type="button"
  >
    Click Me
  </button>
</DocumentFragment>
`;
